dotnet/src/SemanticKernel/SkillDefinition/SKFunction.cs

```diff
@@ -155,44 +155,29 @@ public FunctionView Describe()
 
     /// <inheritdoc/>
     public Task<SKContext> InvokeAsync(
-        string input,
-        SKContext? context = null,
-        CompleteRequestSettings? settings = null,
-        ILogger? log = null,
-        CancellationToken cancellationToken = default)
+        SKContext context,
+        CompleteRequestSettings? settings = null)
     {
-        if (context == null)
-        {
-            log ??= NullLogger.Instance;
-            context = new SKContext(
-                new ContextVariables(""),
-                NullMemory.Instance,
-                this._skillCollection,
-                log,
-                cancellationToken);
-        }
-
-        context.Variables.Update(input);
-
-        return this.InvokeAsync(context, settings, log, cancellationToken);
+        return this.IsSemantic
+            ? this.InvokeSemanticAsync(context, settings)
+            : this.InvokeNativeAsync(context);
     }
 
     /// <inheritdoc/>
     public Task<SKContext> InvokeAsync(
-        SKContext? context = null,
+        string? input = null,
         CompleteRequestSettings? settings = null,
         ILogger? log = null,
         CancellationToken cancellationToken = default)
     {
-        if (context == null)
-        {
-            log ??= NullLogger.Instance;
-            context = new SKContext(new ContextVariables(""), NullMemory.Instance, null, log, cancellationToken);
-        }
-
-        return this.IsSemantic
-            ? this.InvokeSemanticAsync(context, settings)
-            : this.InvokeNativeAsync(context);
+        SKContext context = new(
+            new ContextVariables(input),
+            NullMemory.Instance,
+            this._skillCollection,
+            log ?? NullLogger.Instance,
```

Comment:
Do we expect end-user code to create SKContext instances?  I'm wondering if we should make it easier, e.g. allow the logger to be null and have the ctor pick a default logger if null was passed, or even just allow it to store null.

Reply:
yup, good call